src/libs/ReportUtils.ts

```diff
@@ -20,174 +28,445 @@ import Navigation from './Navigation/Navigation';
+        allPersonalDetails = value ?? {};
     },
 });
 
-let allReports;
+let allReports: OnyxCollection<Report>;
 Onyx.connect({
     key: ONYXKEYS.COLLECTION.REPORT,
     waitForCollectionCallback: true,
     callback: (val) => (allReports = val),
 });
 
-let doesDomainHaveApprovedAccountant;
+let doesDomainHaveApprovedAccountant = false;
 Onyx.connect({
     key: ONYXKEYS.ACCOUNT,
-    waitForCollectionCallback: true,
-    callback: (val) => (doesDomainHaveApprovedAccountant = lodashGet(val, 'doesDomainHaveApprovedAccountant', false)),
+    callback: (value) => (doesDomainHaveApprovedAccountant = value?.doesDomainHaveApprovedAccountant ?? false),
 });
 
-let allPolicies;
+let allPolicies: OnyxCollection<Policy>;
 Onyx.connect({
     key: ONYXKEYS.COLLECTION.POLICY,
     waitForCollectionCallback: true,
     callback: (val) => (allPolicies = val),
 });
 
-let loginList;
+let loginList: OnyxEntry<Login>;
 Onyx.connect({
     key: ONYXKEYS.LOGIN_LIST,
     callback: (val) => (loginList = val),
 });
 
-function getChatType(report) {
-    return report ? report.chatType : '';
+function getChatType(report: OnyxEntry<Report>): ValueOf<typeof CONST.REPORT.CHAT_TYPE> | undefined {
+    return report?.chatType;
 }
 
-/**
- * @param {String} policyID
- * @returns {Object}
- */
-function getPolicy(policyID) {
+function getPolicy(policyID: string): OnyxEntry<Policy> | Record<string, never> {
     if (!allPolicies || !policyID) {
         return {};
     }
-    return allPolicies[`${ONYXKEYS.COLLECTION.POLICY}${policyID}`] || {};
+    return allPolicies[`${ONYXKEYS.COLLECTION.POLICY}${policyID}`] ?? {};
 }
 
 /**
  * Get the policy type from a given report
- * @param {Object} report
- * @param {String} report.policyID
- * @param {Object} policies must have Onyxkey prefix (i.e 'policy_') for keys
- * @returns {String}
+ * @param  policies must have Onyxkey prefix (i.e 'policy_') for keys
  */
-function getPolicyType(report, policies) {
-    return lodashGet(policies, [`${ONYXKEYS.COLLECTION.POLICY}${report.policyID}`, 'type'], '');
+function getPolicyType(report: OnyxEntry<Report>, policies: OnyxCollection<Policy>): string {
+    return policies?.[`${ONYXKEYS.COLLECTION.POLICY}${report?.policyID}`]?.type ?? '';
 }
 
 /**
  * Get the policy name from a given report
- * @param {Object} report
- * @param {String} [report.policyID]
- * @param {String} [report.oldPolicyName]
- * @param {String} [report.policyName]
- * @param {Boolean} [returnEmptyIfNotFound]
- * @param {Object} [policy]
- * @returns {String}
- */
-function getPolicyName(report, returnEmptyIfNotFound = false, policy = undefined) {
+ */
+function getPolicyName(report: OnyxEntry<Report> | undefined, returnEmptyIfNotFound = false, policy: OnyxEntry<Policy> = null): string {
     const noPolicyFound = returnEmptyIfNotFound ? '' : Localize.translateLocal('workspace.common.unavailable');
-    if (_.isEmpty(report)) {
+    if (Object.keys(report ?? {}).length === 0) {
```

Comment:
You could create a util like `isEmptyObject` similar to how you created `isNotEmptyObject`, and use it throughout this file